icu4c/source/i18n/calendar.cpp

```diff
@@ -2764,6 +2772,47 @@ Calendar::inDaylightTime(UErrorCode& status) const
     return (UBool)(U_SUCCESS(status) ? (internalGet(UCAL_DST_OFFSET) != 0) : false);
 }
 
+bool
+Calendar::inTemporalLeapYear(UErrorCode& status) const
+{
+    // Default to Gregorian based leap year rule.
+    return getActualMaximum(UCAL_DAY_OF_YEAR, status) == 366;
+}
+
+// -------------------------------------
+
+static const char * const gTemporalMonthCodes[] = {
+    "M01", "M02", "M03", "M04", "M05", "M06",
+    "M07", "M08", "M09", "M10", "M11", "M12", nullptr
+};
+
+const char*
+Calendar::getTemporalMonthCode(UErrorCode& status) const
+{
+    int32_t month = get(UCAL_MONTH, status);
+    if (U_FAILURE(status)) return nullptr;
+    U_ASSERT(month < 12);
+    U_ASSERT(internalGet(UCAL_IS_LEAP_MONTH) == 0);
```

Comment:
Right, currently, the only subclass of Calendar will set UCAL_IS_LEAP_MONTH is ChineseCalendar which is also a superclass of DangiCalendar. So only ChineseCalendar and DangiCalendar will set that bit. And we handle that in the ChineseCalendar::getTemporalMonthCode but want to ensure in the future if some other subclass of Calendar also set UCAL_IS_LEAP_MONTH we will bark here and force them to add an override of  getTemporalMonthCode()

Reply:
Thanks; I eventually figured that out after reading further into your PR.  It might be worth a quick note here explaining what you're doing (then again, it might not).